index.js

```diff
@@ -506,20 +543,44 @@ function send (res, type, body) {
   res.end(body, 'utf8')
 }
 
+/**
+ * Convert size to human-readable size
+ */
+
+var SIZES = [ 'B', 'K', 'M', 'G', 'T', 'P', 'E' ]
+function hsize(size) {
+  var i = 0
+  while (size > 1024 && i < SIZES.length) {
+    size = Math.floor(size/1024)
+    i += 1
+  }
+  return size + SIZES[i]
+}
+
+
 /**
  * Stat all files and return array of stat
  * in same order.
  */
 
-function stat(dir, files, cb) {
+function fstat(dir, files, cb) {
   var batch = new Batch();
 
   batch.concurrency(10);
 
   files.forEach(function(file){
     batch.push(function(done){
       fs.stat(join(dir, file), function(err, stat){
-        if (err && err.code !== 'ENOENT') return done(err);
+        // communicate errors via fake stat
+        if (err && err.code !== 'ENOENT') {
+          stat = {
+            size: 0,
+            mtime: new Date(0),
+            error: err.toString(),
+            code: err.code,
+            isDirectory: function () { return false }
```

Comment:
I believe your changes will expose this stat object to the users in the previous commits. I assume you made these fake values to prevent code from breaking, but it'll still break since you didn't provide all the values in a native Node.js stat object here. You'll want to make sure all the properties are covered if you're going to provide them at all, otherwise that's going to be even more confusing to the user, imo.

Reply:
I just pushed another commit to the stat-first branch that converts to a w3 FileAPI-esque object sooner, which corrects that issue.

Reply:
Will that show up in this pr at all?

Reply:
No, I made separate branches for each change and only based each change on the fewest number of commits needed to demonstrate the change.

All of the PRs are in order. I’ll go back and add when other PR the depend on.